rust_src/src/frames.rs

```diff
@@ -155,4 +156,27 @@ pub fn framep(object: LispObject) -> LispObject {
     }
 }
 
+/// Return t if FRAME is \"visible\" (actually in use for display).
+/// Return the symbol `icon' if FRAME is iconified or \"minimized\".
+/// Return nil if FRAME was made invisible, via `make-frame-invisible'.
+/// On graphical displays, invisible frames are not updated and are
+/// usually not displayed at all, even in a window system's \"taskbar\".
+///
+/// If FRAME is a text terminal frame, this always returns t.
+/// Such frames are always considered visible, whether or not they are
+/// currently being displayed on the terminal.
+#[lisp_fn]
+pub fn frame_visible_p(frame: LispObject) -> LispObject {
+    let frame_ref = frame.as_live_frame_or_error();
+    if unsafe { fget_visible(frame_ref.as_ptr()) != 0 } {
```

Comment:
go ahead and add the `frame.is_visible()` and `is_iconified` methods and use them here. Then this function should look like the C function.